src/redux/reducers/user.tsx

```diff
@@ -1,53 +1,59 @@
-import {USER_SIGNUP, USER_LOGOUT, USER_UPDATE, AUTHENTICATE, SET_DID_TRY_AL } from '../actions/user'
+import { ActionTypes } from '../actions/user';
 
-export const initialState = {
-  token: null,
-  userId: null,
-  isLoggedIn: false,
-  // firstName: 'Guest',
-  // lastName: 'User',
-  userType: 'consumer',
-  didTryAutoLogin: false
+export const InitialState = {
+  userType: '',
+  uid: '',
+  signUpLoader: false,
+  selected_user_role: '',
+  async_storage_data: {
+    'data': {
+      store: {},
+      isLoggedIn: false
+    }
+  },
 };
-
-export const userReducer = (state = initialState, action) => {
+export default (state = InitialState, action: any) => {
   switch (action.type) {
-    case AUTHENTICATE: {
+    case ActionTypes.USER_SIGN_UP:
       return {
         ...state,
-        isLoggedIn: true, 
-        token: action.token,
-        userId: action.userId,
-        // firstName: action.firstName, 
-        // lastName: action.lastName,
-        userType: action.userType,
-        didTryAutoLogin: true
-      }
-    }
-    case USER_UPDATE: {
+        signUpLoader: action.payload.signupLoader,
+        uid: action.payload.userUID
+      };
+    case ActionTypes.USER_SIGN_IN:
       return {
         ...state,
-        isLoggedIn: true, 
-        token: action.token,
-        userId: action.userId,
-        // firstName: action.firstName, 
-        // lastName: action.lastName, 
-        userType: action.userType,
-        didTryAutoLogin: true
-      }
-    }
-    case SET_DID_TRY_AL:
+        signInLoader: action.payload.signinLoader,
+        uid: action.payload.userUID
+      };
+    case ActionTypes.GET_DATA_FROM_ASYNCSTORAGE:
+      const { role } = action.payload?.data?.store;
       return {
         ...state,
-        userType: action.userType,
-        didTryAutoLogin: true
+        async_storage_data: action.payload,
+        userType: role
       };
-    case USER_LOGOUT: {
+    case ActionTypes.SET_USER_ROLE:
       return {
-        ...initialState,
-        didTryAutoLogin: true
-      }
-    }
-    default: { return state }
+        ...state,
+        userType: action.payload
+      };
+
+    case ActionTypes.SELECT_USER_ROLE:
+      return {
+        ...state,
+        selected_user_role: action.payload
```

Comment:
does async_storage_data and selected_user_role needs to be snake case or can we adjust to camel case, as we use it everywhere else?